---
title: "20211503_DREAM"
author: "Gijsje"
date: "3/15/2021"
output: 
rmarkdown::html_document:
   theme: united
   highlight: tango
   code_folding: hide
   toc: true
   toc_float: true
   df_print: paged
   smooth_scroll: true
   number_sections: false
   self_contained: true
---

```{r setup, include=FALSE}
library('variancePartition')
library('edgeR')
library('BiocParallel')
library(dplyr)
library(tidyr)
```

# Description
#Dream uses a linear model model to increase power and decrease false positives for RNA-seq datasets with repeated measurements.  Dream achieves this by combining multiple statistical concepts into a single statistical model. The model includes: flexible modeling of repeated measures gene expression data, precision weights to model measurement error in RNA-seq counts, ability to model multiple random effects, random effects estimated separately for each gene, hypothesis testing for fixed effects in linear mixed models, small sample size hypothesis test.

```{r kable, echo=FALSE, message=FALSE, warnings=FALSE, results='asis'}
load("~/Documents/MiGASti/Databases/gene_matrix.RData")
metadata <- read.table("~/Documents/MiGASti/Databases/metadata.txt")
#set rownames to Sample
row.names(metadata) <- metadata$Sample 
setwd("~/Documents/MiGASti/Databases")
#exclude samples that did not pass QC filtering
exclude <- read.table("samples2remove.txt")
exclude <- exclude$x
genes_counts_filt = genes_counts[, !colnames(genes_counts) %in% exclude] 
#Excludes the samples from filters. 
#dim(genes_counts_filt)
metadata_filt = metadata[ !(rownames(metadata) %in% exclude), ]
length(metadata_filt)
gencode_30 = read.table("~/Documents/MiGASti/Databases/ens.geneid.gencode.v30")
colnames(gencode_30) = c("ensembl","symbol")
#remove low count genes
cpm <- cpm(genes_counts_filt) 
# CPM >= 1 in at least 50% of the samples
keep.exp <- rowSums(cpm > 1) >= (0.5 * ncol(genes_counts_filt) )
genes_counts_filt1 <- genes_counts_filt[ keep.exp, ] #18997 genes 
#order metadata and genes counts
genes_counts_ordered <- genes_counts_filt1[,rownames(metadata_filt)]
#head(genes_counts_ordered)
all(rownames(metadata_filt) == colnames (genes_counts_ordered)) #TRUE
```

# Limma Analysis
#Limma has a built-in approach for analyzing repeated measures data using duplicatecorrelation. The model can handle a single random effect, and forces the magnitude of the random effect to be the same across all genes.

```{r dupCor, eval=TRUE}
# Standard usage of limma/voom
geneExpr = DGEList(genes_counts_ordered)
geneExpr = calcNormFactors( geneExpr )
```





```{r dupCor, eval=TRUE}
# apply duplicateCorrelation is two rounds
design = model.matrix( ~ Stimulation, metadata_filt)
vobj_tmp = voom( geneExpr, design, plot=FALSE)
dupcor <- duplicateCorrelation(vobj_tmp,design,block=metadata_filt$Donor_id)
# run voom considering the duplicateCorrelation results
# in order to compute more accurate precision weights
# Otherwise, use the results from the first voom run
vobj = voom( geneExpr, design, plot=FALSE, block=metadata_filt$Donor_id, correlation=dupcor$consensus)
# Estimate linear mixed model with a single variance component
# Fit the model for each gene, 
dupcor <- duplicateCorrelation(vobj, design, block=metadata_filt$Donor_id)
# But this step uses only the genome-wide average for the random effect
fitDupCor <- lmFit(vobj, design, block=metadata_filt$Donor_id, correlation=dupcor$consensus)
# Fit Empirical Bayes for moderated t-statistics
fitDupCor <- eBayes( fitDupCor )
head(fitDupCor)
```


# Dream Analysis
#The dream method replaces two core functions of limma with a linear mixed model.
#1. `voomWithDreamWeights()` replaces `voom()` to estimate precision weights 
#2. `dream()` replaces `lmFit()` to estimate regression coefficients.  
#Otherwise dream uses the same workflow as limma with `topTable()`, since any statistical differences are handled behind the scenes.

```{r dupCor, eval=TRUE}
# The variable to be tested must be a fixed effect
form <- ~ Stimulation + (1|Donor_id) 
# estimate weights using linear mixed model of dream
vobjDream = voomWithDreamWeights( geneExpr, form, metadata_filt )
# Fit the dream model on each gene
# By default, uses the Satterthwaite approximation for the hypothesis test
fitmm = dream( vobjDream, form, metadata_filt )
# Examine design matrix
head(fitmm$design, 3)
# Get results of hypothesis test on coefficients of interest
topTable( fitmm, coef='LPS', number=3 )
```

#Since dream uses an estimated degrees of freedom value for each hypothsis test, the degrees of freedom is different for each gene here.Therefore, the t-statistics are not directly comparable since they have different degrees of freedom.  In order to be able to compare test statistics, we report `z.std` which is the p-value transformed into a signed z-score.  This can be used for downstream analysis.

## Advanced hypothesis testing
### Using contrasts to compare coefficients
#You can also perform a hypothesis test of the *difference* between two or more coefficients by using a contrast matrix.  The contrasts are evaluated at the time of the model fit and the results can be extracted with `topTable()`.  This behaves like `makeContrasts()` and `contrasts.fit()` in limma.

#Multiple contrasts can be evaluated at the same time, in order to save computation time:
```{r contrast.combine, eval=TRUE, fig.width=7, fig.height=3}
form <- ~ 0 + Stimulation + Sex + (1|Donor_id) 
# define and then cbind contrasts
L1 = getContrast( vobjDream, form, metadata_filt, c("LPS", "unstim"))
L2 = getContrast( vobjDream, form, metadata_filt, c("IFNY", "unstim"))
L3 = getContrast( vobjDream, form, metadata_filt, c("TNFa", "unstim"))
L4 = getContrast( vobjDream, form, metadata_filt, c("ATP", "unstim"))
L5 = getContrast( vobjDream, form, metadata_filt, c("DEX", "unstim"))
L6 = getContrast( vobjDream, form, metadata_filt, c("R848", "unstim"))
L7 = getContrast( vobjDream, form, metadata_filt, c("IL4", "unstim"))
L = cbind(L1, L2, L3, L4, L5, L6, L7)     
# Visualize contrasts
plotContrasts(L) 
# fit both contrasts
fit = dream( vobjDream, form, metadata, L)
# extract results from first contrast
topTable( fit, coef="L1", number=3 )

```





