---
title: "Genelists_overlap_LPS"
author: "Gijsje"
date: "5/13/2021"
output: 
 rmarkdown::html_document:
   theme: united
   highlight: tango
   code_folding: hide
   toc: true
   toc_float: true
   df_print: paged
   smooth_scroll: true
   number_sections: false
   self_contained: true
---

```{r setup, include=FALSE}
library('variancePartition')
library('edgeR')
library(dplyr)
library(tidyr)
library(readxl)
library(edgeR)
library(limma)
library(DESeq2)
library(tidyverse)
library(data.table)
library(shiny)
library(pheatmap)
library(ggfortify)
library(ggplot2)
library(UpSetR)
library(VennDiagram)
library(knitr) 
library(tidyverse)
library(dplyr)
library(tidyr)
library(tibble)
library(purrr)
library(corrplot)
library(ggplot2)
library(ggpubr)
library(EnhancedVolcano)
library(raster)
library(dplyr)
library(ggrastr)
```

# Enrichment analyses 
#Input
#1. Genelist core microglia signature list n = 249 genes (Patir et al. 2019)
#2. snRNAseq Multiple sclerosis n = 64 genes (Schiermer et al. 2019), 
#3. snRNAseq Alzheimers disesease n = 53 genes (Mathys et al. 2019), 
#4. mouse snRNAseq response to demyelinatin injury after lysolecithin injection into the brain n = 119 genes (Hammond et al. 2019)  
#5. scRNAseq of activation response microglia in AD mouse n = 108 genes (Sala-Frigerio et al. 2019)
#6 scRNAseq disease associated microglia up n = 280 /down n = 22 genes (Keren-Shaul et al. 2019), 
#7. scRNAseq microglia from mouse stimulated with LPS 3 hours n = 367 genes (acute LPS; Gerrits et al. 2020),
#8. scRNAseq microglia from mouse stimulated with LPS 24 hours n = 152 genes (subacute LPS; Sousa et al. 2020), 
#9. GO: NFkB pathway n = 147 genes
#10. GO: IFNy pathway n = 139 genes
#11. GO: cold stress response pathway n = 42 genes
#12. Plaque induced genes n = 57 genes (Grubman et al. 2021)
#13. Sensome genes n = 84 genes (Hickman et al. 2019)
#14. TWAS AD genes (Raj et al. 2018) 
#15. TWAS PD genes (Li et al. 2019)
#16. scRNAseq Human alzheimer genes up n = 22/down n = 53 genes (Srinsivan et al. 2019)
#17. Homeostatic genes up/down n = 44 genes in the mature homeostatic microglia phenotype based on bulk RNAseq Ormel et al. 2018, Gosselin et al. 2017 and Kracht et al. 2020).
#18. bulk RNAseq cultured microglia compared to ex vivo up n = 3674 / down n = 4121 genes (Gosselin et al. 2017)
#19. Immediate early response genes (IEGs), genes whose transcripts are rapidly expressed after cellular stimulation, without the need for new protein synthesis to control this expression n = 128 genes (Wu et al. 2017) 
#20. Microglia genes up n = 338/down n = 1355 in aging (MIGA)

```{r function_setenrichment, echo = TRUE}
# Function set enrichment
setEnrichment <- function(set1, set2, universe = 20000){

  a = sum(set1 %in% set2)

  c = length(set1) - a

  b = length(set2) - a

  d = universe - length(set2) - c

  contingency_table = matrix(c(a, c, b, d), nrow = 2)

  # one-tailed test for enrichment only

  fisher_results = fisher.test(contingency_table, alternative = "greater")

  # returns data.frame containing the lengths of the two sets, the overlap, the enrichment ratio (odds ratio) and P value

  df <- tibble::tibble( set1_length = length(set1), set2_length = length(set2), overlap = a, ratio = fisher_results$estimate, p.value = fisher_results$p.value)

  return(df)
}
```

# Data load

```{r Patir, echo = TRUE}
Patir <- read.csv("~/Documents/MiGASti/Databases/Patir.txt", sep="")
names(Patir)[1] <- "symbol"

ARM = "~/Documents/MiGASti/Databases/Activation_response_microglia.xlsx"
ARM = read_excel(ARM, col_names = TRUE) 
ARM = as.data.frame(ARM)

IRM = "~/Documents/MiGASti/Databases/Immediate_response_microglia.xlsx"
IRM = read_excel(IRM, col_names = TRUE) 
IRM = as.data.frame(IRM)

NFkB = "~/Documents/MiGASti/Databases/NfkB_activation.xlsx"
NFkB = read_excel(NFkB, col_names = TRUE) 
NFkB = as.data.frame(NFkB)

Cold = "~/Documents/MiGASti/Databases/Cold.xlsx"
Cold = read_excel(Cold, col_names = TRUE) 
Cold = as.data.frame(Cold)

SA_LPS = "~/Documents/MiGASti/Databases/subacute_microglia.xlsx"
SA_LPS = read_excel(SA_LPS, col_names = TRUE) 
SA_LPS = as.data.frame(SA_LPS)

A_LPS = "~/Documents/MiGASti/Databases/acute_LPS.xlsx"
A_LPS = read_excel(A_LPS, col_names = TRUE) 
A_LPS = as.data.frame(A_LPS)

InjRM = "~/Documents/MiGASti/Databases/Injury_responsive.xlsx"
InjRM = read_excel(InjRM, col_names = TRUE) 
InjRM = as.data.frame(InjRM)

AD = "~/Documents/MiGASti/Databases/Alzheimers_disease.xlsx"
AD = read_excel(AD, col_names = TRUE) 
AD = as.data.frame(AD)

MS = "~/Documents/MiGASti/Databases/Multiple_sclerosis.xlsx"
MS = read_excel(MS, col_names = TRUE) 
MS = as.data.frame(MS)

INV_down = "~/Documents/MiGASti/Databases/INVITRO_DOWN.xlsx"
INV_down = read_excel(INV_down, col_names = TRUE) 
INV_down = as.data.frame(INV_down)

DAM_down = "~/Documents/MiGASti/Databases/DAM_DOWN.xlsx"
DAM_down = read_excel(DAM_down, col_names = TRUE) 
DAM_down = as.data.frame(DAM_down)

DAM_up = "~/Documents/MiGASti/Databases/DAM_up.xlsx"
DAM_up = read_excel(DAM_up, col_names = TRUE) 
DAM_up = as.data.frame(DAM_up)

HAM_up = "~/Documents/MiGASti/Databases/HAM_up.xlsx"
HAM_up = read_excel(HAM_up, col_names = TRUE) 
HAM_up = as.data.frame(HAM_up)

HAM_down = "~/Documents/MiGASti/Databases/HAM_DOWN.xlsx"
HAM_down = read_excel(HAM_down, col_names = TRUE) 
HAM_down = as.data.frame(HAM_down)

INV_up = "~/Documents/MiGASti/Databases/INVITRO_UP.xlsx"
INV_up = read_excel(INV_up, col_names = TRUE) 
INV_up = as.data.frame(INV_up)

TWAS_PD = "~/Documents/MiGASti/Databases/TWAS_PD.xlsx"
TWAS_PD = read_excel(TWAS_PD, col_names = TRUE) 
TWAS_PD = as.data.frame(TWAS_PD)

TWAS_AD = "~/Documents/MiGASti/Databases/TWAS_AD.xlsx"
TWAS_AD = read_excel(TWAS_AD, col_names = TRUE) 
TWAS_AD = as.data.frame(TWAS_AD)

INFy_GO = "~/Documents/MiGASti/Databases/IFNy_GO.xlsx"
INFy_GO = read_excel(INFy_GO, col_names = TRUE) 
INFy_GO = as.data.frame(INFy_GO)

PIG = "~/Documents/MiGASti/Databases/PIG.xlsx"
PIG = read_excel(PIG, col_names = TRUE) 
PIG = as.data.frame(PIG)

Sensome = "~/Documents/MiGASti/Databases/Sensome_human.xlsx"
Sensome = read_excel(Sensome, col_names = TRUE) 
Sensome = as.data.frame(Sensome)

aging_down = "~/Documents/MiGASti/Databases/aging_down_MIGA.xlsx"
aging_down = read_excel(aging_down, col_names = TRUE) 
aging_down = as.data.frame(aging_down)

aging_up = "~/Documents/MiGASti/Databases/aging_up_MIGA.xlsx"
aging_up = read_excel(aging_up, col_names = TRUE) 
aging_up = as.data.frame(aging_up)

HOM = "~/Documents/MiGASti/Databases/Homeostatic_genes.xlsx"
HOM = read_excel(HOM, col_names = TRUE) 
HOM = as.data.frame(HOM)

load("~/Documents/MiGASti/docs/2nd_pass/AD_bulk_3datasets.Rdata")
AD_bulk <- genes3
```

# LPS UP FDR 5% logFC > 1 

```{r vul_LPS, echo = TRUE}
load("~/Documents/MiGASti/docs/2nd_pass/res_LPS_name.Rdata")
res_LPS <- res_LPS_name

sign <- subset(res_LPS, adj.P.Val < 0.05)
LPS_UP <- subset(sign, logFC > 1)

set1 <- LPS_UP
set1_v <- set1$symbol
```

### + Patir

```{r LPS_Patir, echo = TRUE}
set2 <- Patir
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute responsive microglia 

```{r LPS_ARM, echo = TRUE}
set2 <- ARM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + immediate responsive microglia 

```{r LPS_IRM, echo = TRUE}
set2 <- IRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + NFkB GO 

```{r LPS_NFkB, echo = TRUE}
set2 <- NFkB
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INFy GO

```{r LPS_INFy_GO, echo = TRUE}
set2 <- INFy_GO
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Cold 

```{r LPS_COLD, echo = TRUE}
set2 <- Cold
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute_LPS 

```{r LPS_acute, echo = TRUE}
set2 <- A_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + subacute_LPS 

```{r LPS_subacute, echo = TRUE}
set2 <- SA_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Injury responsive microglia 

```{r LPS_InjRM, echo = TRUE}
set2 <- InjRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Plaque induced genes 

```{r LPS_PIG, echo = TRUE}
set2 <- PIG
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Sensome genes 

```{r LPS_Sensome, echo = TRUE}
set2 <- Sensome
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes mouse 

```{r LPS_AD, echo = TRUE}
set2 <- AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS AD genes

```{r LPS_AD_TWAS, echo = TRUE}
set2 <- TWAS_AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes bulk brain 

```{r LPS_AD_bulk5, echo = TRUE}
set2 <- AD_bulk
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + MS

```{r LPS_MS, echo = TRUE}
set2 <- MS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS PD

```{r LPS_TWAS_PD, echo = TRUE}
set2 <- TWAS_PD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_up

```{r LPS_HAM_up, echo = TRUE}
set2 <- HAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_down

```{r LPS_HAM_down, echo = TRUE}
set2 <- HAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_up

```{r LPS_DAM_up, echo = TRUE}
set2 <- DAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_down

```{r LPS_DAM_down, echo = TRUE}
set2 <- DAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_up

```{r LPS_INV_up, echo = TRUE}
set2 <- INV_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_down

```{r LPS_INV_down, echo = TRUE}
set2 <- INV_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_up

```{r LPS_aging_up, echo = TRUE}
set2 <- aging_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_down

```{r LPS_aging_down, echo = TRUE}
set2 <- aging_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + homeostatic genes

```{r LPS_HOM2, echo = TRUE}
set2 <- HOM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

# LPS DOWN FDR 5% logFC < -1 

```{r vul_LPS2, echo = TRUE}
load("~/Documents/MiGASti/res_name_LPS2.Rdata")
res_LPS <- res_name_LPS2

sign <- subset(res_LPS, adj.P.Val < 0.05)
LPS_DOWN <- subset(sign, logFC < -1)

set1 <- LPS_DOWN
set1_v <- set1$symbol
```

### + Patir

```{r LPS_Patir2, echo = TRUE}
set2 <- Patir
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute responsive microglia 

```{r LPS_ARM2, echo = TRUE}
set2 <- ARM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + immediate responsive microglia 

```{r LPS_IRM2, echo = TRUE}
set2 <- IRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + NFkB GO 

```{r LPS_NFkB2, echo = TRUE}
set2 <- NFkB
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INFy GO

```{r LPS_INFy2, echo = TRUE}
set2 <- INFy_GO
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Cold 

```{r LPS_COLD2, echo = TRUE}
set2 <- Cold
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute_LPS 

```{r LPS_acute2, echo = TRUE}
set2 <- A_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + subacute_LPS 

```{r LPS_subacute2, echo = TRUE}
set2 <- SA_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Injury responsive microglia 

```{r LPS_InjRM2, echo = TRUE}
set2 <- InjRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Plaque induced genes 

```{r LPS_PIG2, echo = TRUE}
set2 <- PIG
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Sensome genes 

```{r LPS_Sensome2, echo = TRUE}
set2 <- Sensome
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes mouse 

```{r LPS_AD2, echo = TRUE}
set2 <- AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes bulk brain 

```{r LPS_AD_bulk2, echo = TRUE}
set2 <- AD_bulk
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS AD genes

```{r LPS_AD_TWAS2, echo = TRUE}
set2 <- TWAS_AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + MS

```{r LPS_MS2, echo = TRUE}
set2 <- MS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS PD

```{r LPS_TWAS_PD2, echo = TRUE}
set2 <- TWAS_PD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_up

```{r LPS_HAM_up2, echo = TRUE}
set2 <- HAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_down

```{r LPS_HAM_down2, echo = TRUE}
set2 <- HAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_up

```{r LPS_DAM_up2, echo = TRUE}
set2 <- DAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_down

```{r LPS_DAM_down2, echo = TRUE}
set2 <- DAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_up

```{r LPS_INV_up2, echo = TRUE}
set2 <- INV_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_down

```{r LPS_INV_down2, echo = TRUE}
set2 <- INV_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_up

```{r LPS_aging_up2, echo = TRUE}
set2 <- aging_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_down

```{r LPS_aging_down2, echo = TRUE}
set2 <- aging_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + homeostatic genes

```{r LPS_HOM3, echo = TRUE}
set2 <- HOM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

# LPS UP FDR 5% 

```{r vul_LPS3, echo = TRUE}
load("~/Documents/MiGASti/res_name_LPS2.Rdata")
res_LPS <- res_name_LPS2

sign <- subset(res_LPS, adj.P.Val < 0.05)
LPS_UP <- subset(sign, logFC > 0)

set1 <- LPS_UP

set1 <- LPS_UP
set1_v <- set1$symbol
```

### + Patir

```{r LPS_Patir3, echo = TRUE}
set2 <- Patir
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute responsive microglia 

```{r LPS_ARM3, echo = TRUE}
set2 <- ARM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + immediate responsive microglia 

```{r LPS_IRM3, echo = TRUE}
set2 <- IRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + NFkB GO 

```{r LPS_NFkB3, echo = TRUE}
set2 <- NFkB
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INFy GO

```{r LPS_INFy3, echo = TRUE}
set2 <- INFy_GO
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Cold 

```{r LPS_COLD3, echo = TRUE}
set2 <- Cold
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute_LPS 

```{r LPS_acute3, echo = TRUE}
set2 <- A_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + subacute_LPS 

```{r LPS_subacute3, echo = TRUE}
set2 <- SA_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Injury responsive microglia 

```{r LPS_InjRM3, echo = TRUE}
set2 <- InjRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Plaque induced genes 

```{r LPS_PIG3, echo = TRUE}
set2 <- PIG
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Sensome genes 

```{r LPS_Sensome3, echo = TRUE}
set2 <- Sensome
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes mouse 

```{r LPS_AD3, echo = TRUE}
set2 <- AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes bulk brain 

```{r LPS_AD_bulk, echo = TRUE}
set2 <- AD_bulk
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS AD genes

```{r LPS_AD_TWAS3, echo = TRUE}
set2 <- TWAS_AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + MS

```{r LPS_MS3, echo = TRUE}
set2 <- MS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS PD

```{r LPS_TWAS_PD3, echo = TRUE}
set2 <- TWAS_PD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_up

```{r LPS_HAM_up3, echo = TRUE}
set2 <- HAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_down

```{r LPS_HAM_down3, echo = TRUE}
set2 <- HAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_up

```{r LPS_DAM_up3, echo = TRUE}
set2 <- DAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_down

```{r LPS_DAM_down3, echo = TRUE}
set2 <- DAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_up

```{r LPS_INV_up3, echo = TRUE}
set2 <- INV_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_down

```{r LPS_INV_down3, echo = TRUE}
set2 <- INV_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_up

```{r LPS_aging_up3, echo = TRUE}
set2 <- aging_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_down

```{r LPS_aging_down3, echo = TRUE}
set2 <- aging_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```


### + homeostatic genes

```{r LPS_HOM4, echo = TRUE}
set2 <- HOM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

# LPS DOWN FDR 5%  

```{r vul_LPS4, echo = TRUE}
load("~/Documents/MiGASti/res_name_LPS2.Rdata")
res_LPS <- res_name_LPS2

sign <- subset(res_LPS, adj.P.Val < 0.05)
LPS_DOWN <- subset(sign, logFC < 0)

set1 <- LPS_DOWN
set1_v <- set1$symbol
```

### + Patir

```{r LPS_Patir4, echo = TRUE}
set2 <- Patir
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute responsive microglia 

```{r LPS_ARM4, echo = TRUE}
set2 <- ARM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + immediate responsive microglia 

```{r LPS_IRM4, echo = TRUE}
set2 <- IRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + NFkB GO 

```{r LPS_NFkB4, echo = TRUE}
set2 <- NFkB
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INFy GO

```{r LPS_INFy4, echo = TRUE}
set2 <- INFy_GO
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Cold 

```{r LPS_COLD4, echo = TRUE}
set2 <- Cold
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + acute_LPS 

```{r LPS_acute4, echo = TRUE}
set2 <- A_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + subacute_LPS 

```{r LPS_subacute4, echo = TRUE}
set2 <- SA_LPS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Injury responsive microglia 

```{r LPS_InjRM4, echo = TRUE}
set2 <- InjRM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Plaque induced genes 

```{r LPS_PIG4, echo = TRUE}
set2 <- PIG
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + Sensome genes 

```{r LPS_Sensome4, echo = TRUE}
set2 <- Sensome
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes mouse 

```{r LPS_AD4, echo = TRUE}
set2 <- AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + AD genes bulk brain 

```{r LPS_AD_bulk4, echo = TRUE}
set2 <- AD_bulk
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS AD genes

```{r LPS_AD_TWAS4, echo = TRUE}
set2 <- TWAS_AD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + MS

```{r LPS_MS4, echo = TRUE}
set2 <- MS
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + TWAS PD

```{r LPS_TWAS_PD4, echo = TRUE}
set2 <- TWAS_PD
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_up

```{r LPS_HAM_up4, echo = TRUE}
set2 <- HAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + HAM_down

```{r LPS_HAM_down4, echo = TRUE}
set2 <- HAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_up

```{r LPS_DAM_up4, echo = TRUE}
set2 <- DAM_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + DAM_down

```{r LPS_DAM_down4, echo = TRUE}
set2 <- DAM_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_up

```{r LPS_INV_up4, echo = TRUE}
set2 <- INV_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + INV_down

```{r LPS_INV_down4, echo = TRUE}
set2 <- INV_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_up

```{r LPS_aging_up4, echo = TRUE}
set2 <- aging_up
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + aging_down

```{r LPS_aging_down4, echo = TRUE}
set2 <- aging_down
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```

### + homeostatic genes

```{r LPS_HOM5, echo = TRUE}
set2 <- HOM
set2_v <- set2$symbol
setEnrichment( set1 = set1_v, set2 = set2_v)
overlap <- merge(set1, set2, by = "symbol")
list(overlap$symbol)
```







