---
title: "20210412_DTU"
output: 
rmarkdown::html_document:
   theme: united
   highlight: tango
   code_folding: hide
   toc: true
   toc_float: true
   df_print: paged
   smooth_scroll: true
   number_sections: false
   self_contained: true 
---

```{r setup, include=FALSE}
library(tidyverse)
library(SummarizedExperiment)
library(edgeR)
library(readr)
load("~/MiGASti/tx_matrix.Rdata")
metadata <- read.table("~/MiGASti/metadata.txt", quote="\"", comment.char="")
#set rownames to Sample
row.names(metadata) <- metadata$Sample
names(tx_counts) <- gsub(".isoforms.results", "", names(tx_counts))
#exclude samples that did not pass QC filtering
exclude <- read.csv("~/MiGASti/samples2remove.txt", sep="")
exclude <- exclude$x
tx_counts_filt = tx_counts[, !colnames(tx_counts) %in% exclude] 
#Excludes the samples from filters. 
#dim(genes_counts_filt)
metadata_filt = metadata[ !(rownames(metadata) %in% exclude), ]
length(metadata_filt)
#remove low count genes #208688 transcripts
cpm <- cpm(tx_counts_filt) 
# CPM >= 1 in at least 50% of the samples
keep.exp <- rowSums(cpm > 1) >= (0.5 * ncol(tx_counts_filt) )
tx_counts_filt1 <- tx_counts_filt[ keep.exp, ] #35070 transcripts 
#order metadata and genes counts
tx_counts_ordered <- tx_counts_filt1[,rownames(metadata_filt)]
#head(genes_counts_ordered)
all(rownames(metadata_filt) == colnames (tx_counts_ordered)) #TRUE

#remove uncultured samples
metadata_cultured <- metadata_filt[metadata_filt$Stimulation != "ununstim",]
#check numbers
#dim(metadata_cultured)
#check numbers per stimulation
#table(metadata_filt$Stimulation)
tx_counts_cultured <- tx_counts_ordered[,metadata_cultured$Sample]
#order metadata and genes_counts
tx_counts_cultured <- tx_counts_cultured[,rownames(metadata_cultured)]
#check ordering
all(rownames(metadata_cultured) == colnames (tx_counts_cultured)) #TRUE
gene_meta <- 
  read_tsv("/sc/arion/projects/als-omics/microglia_stimulated/gencode.v30.tx2gene.tsv") %>% 
  janitor::clean_names() %>%
  rename(txname = "isoform_id", geneid = "gene_id", genename = "gene")

table(metadata_cultured$Sample) %in% names(tx_counts_cultured)

#metadata$Sample %in% names(tx_counts)
tx_counts_cultured <- tx_counts_cultured[,metadata_cultured$Sample]

txInfo <- gene_meta %>%
  select(-gene) %>%
  as.data.frame()

rownames(txInfo) <- txInfo$isoform_id

txInfo <- txInfo[txInfo$isoform_id %in% rownames(tx_counts_cultured), ]
txInfo <- subset(txInfo, 
                 duplicated(gene_id) | duplicated(gene_id, fromLast = TRUE))
tx_counts_cultured <- tx_counts_cultured[which(
  rownames(tx_counts_cultured) %in% txInfo$isoform_id), ]

# # filtering
#filter_edgeR <- edgeR::filterByExpr(tx_counts_cultured,
#                              group = support$condition,
#                              min.count = 10,
#                              min.total.count = 10 * 255 * 0.8,
#                              large.n = 50,
#                              min.prop = 0.8) 


# Update txInfo according to the filtering procedure
txInfo <- txInfo[which(
  txInfo$isoform_id %in% rownames(tx_counts_cultured)), ]

# remove txs that are the only isoform expressed within a gene (after filtering)
txInfo <- subset(txInfo, 
                 duplicated(gene_id) | duplicated(gene_id, fromLast = TRUE))
tx_counts_cultured <- tx_counts_cultured[which(rownames(
  tx_counts_cultured) %in% txInfo$isoform_id), ] #25905 transcripts

# for testing
#tx_counts_cultured <- tx_counts_cultured[1:5000,]
tx_counts_cultured <- floor(tx_counts_cultured)


# satuRn requires the transcripts in the rowData and 
# the transcripts in the count matrix to be in the same order.
txInfo <- txInfo[match(rownames(tx_counts_cultured), txInfo$isoform_id), ]


# calculate transcript ratios for all transcripts tested
tx_ratio <- 
  map_df(unique(txInfo$gene_id),  ~{
    df <- tx_counts_cultured[ txInfo$gene_id == .x,]
    ratios <- sweep(df, MARGIN = 2, STATS =  colSums(df), FUN = "/")
    return(ratios)
  })


length(unique(txInfo$gene_id))
# 7449 genes
# 25905 isoforms

## Generate SummarizedExperiment
sumExp <- SummarizedExperiment::SummarizedExperiment(
  assays = list(counts = tx_counts_cultured),
  colData = metadata_cultured,
  rowData = txInfo
)


# specify design formula from colData
metadata(sumExp)$formula <- ~ 0 + as.factor(colData(sumExp)$Stimulation) + as.factor(colData(sumExp)$Region) + colData(sumExp)$picard_pct_mrna_bases + 
  + colData(sumExp)$picard_pct_ribosomal_bases + as.factor(colData(sumExp)$Sex) + colData(sumExp)$Age + colData(sumExp)$picard_pct_percent_duplication + colData(sumExp)$picard_pct_pf_reads_aligned


#sumExp

# fit model - takes a few minutes
system.time({
  sumExp <- satuRn::fitDTU(
    object = sumExp,
    formula = ~ 0 + sumExp$Stimulation + sumExp$picard_pct_mrna_bases + 
      sumExp$picard_pct_pf_reads_aligned + sumExp$picard_pct_ribosomal_bases + 
      sumExp$sex + sumExp$age + sumExp$picard_pct_percent_duplication + sumExp$Region,
    parallel = FALSE,
    BPPARAM = BiocParallel::bpparam(),
    verbose = TRUE
  )
})


# create contrasts
group <- as.factor(metadata_cultured$Stimulation)
design <- model.matrix(~ 0 + sumExp$Stimulation + sumExp$picard_pct_mrna_bases + sumExp$Region +
      sumExp$picard_pct_pf_reads_aligned + sumExp$picard_pct_ribosomal_bases + 
      sumExp$sex + sumExp$age + sumExp$picard_pct_percent_duplication + sumExp$Region) # construct design matrix

colnames(design) <- c("StimulationATP", "StimulationDEX", "StimulationIFNy", "StimulationIL4", "StimulationLPS", "StimulationR848", "StimulationTNFa", "Stimulationunstim", "picard_pct_mrna_bases", "RegionGFM", "RegionGFS", "RegionGTS", "RegionSVZ", "RegionTHA", "picard_pct_pf_reads_aligned", "picard_pct_ribosomal_bases", "sexm", "age", "picard_pct_percent_duplication")

L <- limma::makeContrasts(
  Contrast1 = Stimulationunstim - StimulationLPS,
  Contrast2 = Stimulationunstim - StimulationDEX,
  Contrast3 = Stimulationunstim - StimulationIFNy,
  Contrast4 = Stimulationunstim - StimulationIL4,
  Contrast5 = Stimulationunstim - StimulationR848,
  Contrast6 = Stimulationunstim - StimulationTNFa,
  Contrast7 = Stimulationunstim - StimulationATP,
  age = age,
  levels = design
)

# perform DTU
sumExp <- satuRn::testDTU(
  object = sumExp,
  contrasts = L,
  plot = FALSE,
  sort = TRUE
)


all_res <- 
  list(
  LPS = rowData(sumExp)[["fitDTUResult_Contrast1"]],
  DEX = rowData(sumExp)[["fitDTUResult_Contrast2"]],
  IFNy = rowData(sumExp)[["fitDTUResult_Contrast3"]],
  IL4 = rowData(sumExp)[["fitDTUResult_Contrast4"]],
  R848 = rowData(sumExp)[["fitDTUResult_Contrast5"]],
  TNFa = rowData(sumExp)[["fitDTUResult_Contrast6"]],
  ATP = rowData(sumExp)[["fitDTUResult_Contrast7"]],
  age = rowData(sumExp)[["fitDTUResult_age"]]
  ) %>%
  map( ~{
      rownames_to_column(.x, var = "isoform_id") %>% 
      left_join(gene_meta, by = "isoform_id") %>%
      tibble() }
      )

#save(sumExp, all_res, tx_ratio, file = "splicing/dtu/microglia_stimulation_dtu.RData")
```

