---
title: "QC"
author: "Gijsje"
date: "2/3/2021"
output:
rmarkdown::html_document:
   theme: united
   highlight: tango
   code_folding: hide
   toc: true
   toc_float: true
   df_print: paged
   smooth_scroll: true
   number_sections: false
   self_contained: true 
---

```{r load.packages, echo=FALSE, message=FALSE, results='hide'}
library("readxl") #to read excel into R
library(knitr)
library(dplyr)
library(ggplot2)
library(RColorBrewer)
library(UpSetR)
require(grid)
require(gridExtra)
library(ggfortify)
library(ggsci)
library(readxl)
library(factoextra)
library(ggpubr)
library(data.table)


knitr::opts_chunk$set( 
  warning=FALSE,
  message=FALSE,
  results = 'asis',
  error = FALSE,
  tidy = FALSE,
  fig.show = "hold")
```

```{r folders, echo=TRUE}
setwd("~/Documents/MiGASti/Databases")
multi_qc_data = "~/Documents/MiGASti/Databases/multiqc_general_stats_533s.txt"
general_stats <- read.table("multiqc_general_stats_533s.txt", header=T, stringsAsFactors = FALSE)
multiqc_picard_coverage = read.table("mqc_picard_rna_coverage_1.txt", header = T, stringsAsFactors = F, check.names = F, sep = "\t", row.names = 1)
picard_rnaseqmetrics = read.table("mqc_picard_rnaseqmetrics_assignment_plot_1.txt", header = T, stringsAsFactors = F, check.names = F, sep = '\t')
```

### Mean aligned uniquely mapped reads : 27823519

```{r mapped.reads_mean, echo=TRUE}
mean_aligned = mean(general_stats$STAR_mqc.generalstats.star.uniquely_mapped, na.rm = T)
mean_aligned
```

### Median aligned uniquely mapped reads: 2782664

```{r mapped.reads_median, echo=TRUE}
median_aligned = median(general_stats$STAR_mqc.generalstats.star.uniquely_mapped, na.rm = T)
median_aligned
```

### Maximum aligned uniquely mapped reads: 127595848

```{r mapped.reads_max, echo=TRUE}
max_aligned = max(general_stats$STAR_mqc.generalstats.star.uniquely_mapped, na.rm = T)
na.omit(general_stats[general_stats$STAR_mqc.generalstats.star.uniquely_mapped == max_aligned, c("Sample", "STAR_mqc.generalstats.star.uniquely_mapped")]) 
```

### Minimum aligned uniquely mapped reads:    64850

```{r mapped.reads_min, echo=TRUE}
min_aligned = min(general_stats$STAR_mqc.generalstats.star.uniquely_mapped , na.rm = T)
na.omit(general_stats[general_stats$STAR_mqc.generalstats.star.uniquely_mapped == min_aligned, c("Sample", "STAR_mqc.generalstats.star.uniquely_mapped")]) 
```

### Distrubution plot aligned uniquely mapped reads

```{r mapped.reads_plot, echo=TRUE, fig.width=6, fig.height=6, dpi=300}
ggplot(general_stats, aes(x=STAR_mqc.generalstats.star.uniquely_mapped)) +
  geom_histogram(bins = 100,
                 colour="black", fill="white", xlab="test") +
  labs(x="Uniquely mapped reads", y="Number of samples") +
  geom_vline(xintercept = 10000000, linetype="dotted", color="red", size=0.7) + 
  scale_y_continuous(breaks = (1:100))
```

### Mapped reads  * (1- percent duplication) 

```{r mapped.reads_duplicates, echo=TRUE, fig.width=6, fig.height=6, dpi=300}
mapped_duplicates = general_stats$STAR_mqc.generalstats.star.uniquely_mapped * (1- general_stats$Picard_mqc.generalstats.picard.PERCENT_DUPLICATION) 
ggplot() + geom_histogram(aes(mapped_duplicates))
median(mapped_duplicates)
sd(mapped_duplicates)
```


### STAR < 10 million reads 

```{r STAR_reads10, echo=TRUE}
star_alignment_10m = general_stats$Sample[which(general_stats$STAR_mqc.generalstats.star.uniquely_mapped < 10000000)]
length(star_alignment_10m)
unique(star_alignment_10m)
```

### STAR < 5 million reads 

```{r STAR_5, echo=TRUE}
star_alignment_5m = general_stats$Sample[which(general_stats$STAR_mqc.generalstats.star.uniquely_mapped < 5000000)]
length(star_alignment_5m)
unique(star_alignment_5m)
```

### STAR < 1 million reads

```{r STAR_reads_5m, echo=TRUE}
star_alignment_1m = general_stats$Sample[which(general_stats$STAR_mqc.generalstats.star.uniquely_mapped < 1000000)]
length(star_alignment_1m)
unique(star_alignment_1m)
```

### Ribosomal bases > 30%

```{r ribosomal_bases_30, echo=TRUE}
ribosomal_bases_30  = general_stats$Sample[which(general_stats$Picard_mqc.generalstats.picard.PCT_RIBOSOMAL_BASES > 30)]
as.data.frame(ribosomal_bases_30)
```

### Ribosomal bases > 20%

```{r ribosomal_bases_20, echo=TRUE}
ribosomal_bases_20 = general_stats$Sample[which(general_stats$Picard_mqc.generalstats.picard.PCT_RIBOSOMAL_BASES > 20)]
as.data.frame(ribosomal_bases_20)
```

### Distribution plot PICARD: % ribosomal RNA

```{r PICARD_ribosomal, echo=TRUE, fig.width=6, fig.height=6, dpi=300}
ggplot(general_stats, aes(x=Picard_mqc.generalstats.picard.PCT_RIBOSOMAL_BASES)) +
  geom_histogram(bins = 100,
                 colour="black", fill="white", xlab="test") +
  labs(x="Percentage of rRNA", y="Number of samples") +
  geom_vline(xintercept = 20, linetype="dotted", color="red", size=0.7) + 
  scale_y_continuous(breaks = (1:100))
```

### mRNA mapping < 10%

```{r mRNA_mapping_10, echo=TRUE}
mrna_mapping_10 = general_stats$Sample[which(general_stats$Picard_mqc.generalstats.picard.PCT_MRNA_BASES < 10)]
as.data.frame(mrna_mapping_10)
```

### mRNA mapping < 5%

```{r mRNA_mapping_5, echo=TRUE}
mrna_mapping_5 = general_stats$Sample[which(general_stats$Picard_mqc.generalstats.picard.PCT_MRNA_BASES < 5)]
as.data.frame(mrna_mapping_5)
```

### Distribution plot PICARD: < 5% coding regions 

```{r PICARD_coding_5, echo=TRUE, fig.width=6, fig.height=6, dpi=300}
ggplot(general_stats, aes(x=Picard_mqc.generalstats.picard.PCT_MRNA_BASES)) +
  geom_histogram(bins = 100,
                 colour="black", fill="white", xlab="test") +
  labs(x="Percent of aligned bases overlapping UTRs and coding regions", y="Number of samples") +
  geom_vline(xintercept = 5, linetype="dotted", color="red", size=0.7) + 
  scale_y_continuous(breaks = (1:100))
```

### Distribution plot PICARD: 10% coding regions

```{r PICARD_coding_10, echo=TRUE, fig.width=6, fig.height=6, dpi=300}
ggplot(general_stats, aes(x=Picard_mqc.generalstats.picard.PCT_MRNA_BASES)) +
  geom_histogram(bins = 100,
                 colour="black", fill="white", xlab="test") +
  labs(x="Percent of aligned bases overlapping UTRs and coding regions", y="Number of samples") +
  geom_vline(xintercept = 10, linetype="dotted", color="red", size=0.7) + 
  scale_y_continuous(breaks = (1:100))
```

### Gene coverage plot 

```{r gene_coverage_all, echo=TRUE, fig.width=6, fig.height=6, dpi=300}

#check dataframe
multiqc_picard_coverage[1:10, 1:6]

#create plot 
x = colnames(multiqc_picard_coverage)
x = as.numeric(x)
colfunc <- colorRampPalette(c("#4DBBD5FF", "#3C5488FF")) # Blue colors
colors = alpha(colfunc(nrow(multiqc_picard_coverage)), alpha = 0.5) # Ajuste o alpha pra ter mais ou menos transparencia
y = multiqc_picard_coverage[1,]
plot(x, y, type = "l", xlab = "Gene body", ylab = "Read coverage", col=colors[1], ylim = c(0,2))
for (i in 2:nrow(multiqc_picard_coverage))
{
  y = multiqc_picard_coverage[i,]
  lines(x,y, col=colors[i])
}
```

### Make table with samples 95 position (degraded: '3 end)

```{r gene_coverage_95, echo=TRUE}
cov_95 = multiqc_picard_coverage$'95'
cov_95 = as.data.frame(cov_95)
rownames(cov_95) = rownames(multiqc_picard_coverage)
colnames(cov_95) = c("position_95")
cov_95_ordered = cov_95[order(cov_95$position_95, decreasing = T),,drop=F]
data.table(cov_95_ordered)
```

### Show list of samples 95 coverage 

```{r gene_coverage_95_exclude, echo=TRUE}
cov_95 = multiqc_picard_coverage$'95'
cov_95 = as.data.frame(cov_95)
rownames(cov_95) = rownames(multiqc_picard_coverage)
colnames(cov_95) = c("position_95")
cov_95_ordered = cov_95[order(cov_95$position_95, decreasing = T),,drop=F]
data.table(cov_95_ordered)
```

### Exclude samples 95 coverage 
526 position 95 1.26 MG-24-HIP-RNA
494 position 95 1.12 MG-12-CER-RNA
130 position 95 1.06 15-089-GTS-unstim
503 position 95 1.04 MG-15-SVZ-RNA
492 position 95 1.04 MG-08-CER-RNA

```{r gene_coverage_exclude, echo=TRUE}
exclude = rownames(cov_95_ordered)[1:6]
multiqc_picard_coverage_filt = multiqc_picard_coverage[! rownames(multiqc_picard_coverage) %in% exclude ,]
```

### Make gene coverage plot again 

```{r gene_coverage_subsample, echo=TRUE, fig.width=6, fig.height=6, dpi=300}
x = colnames(multiqc_picard_coverage_filt)
x = as.numeric(x)
colfunc <- colorRampPalette(c("#4DBBD5FF", "#3C5488FF")) # Blue colors
colors = alpha(colfunc(nrow(multiqc_picard_coverage_filt)), alpha = 0.5) # Ajuste o alpha pra ter mais ou menos transparencia
y = multiqc_picard_coverage_filt[1,]
plot(x, y, type = "l", xlab = "Gene body", ylab = "Read coverage", col=colors[1], ylim = c(0,2))
for (i in 2:nrow(multiqc_picard_coverage_filt))
{
  y = multiqc_picard_coverage_filt[i,]
  lines(x,y, col=colors[i])
}
```

### Samples with < 5% coding regions

```{r coding_regions_5, echo=TRUE}
coding_mapping = picard_rnaseqmetrics[,"Coding"] / rowSums(picard_rnaseqmetrics[,-1])
names(coding_mapping) = picard_rnaseqmetrics$Sample
filter_coding = coding_mapping < 0.05 

names(filter_coding)[which(filter_coding == TRUE)]
data.table(filter_coding)
```

### Filters
# Filter 1 < 5 M mapped reads: 21 samples
# Filter 2 mRNA mapping < 5%: 11 samples
# Filter 3 Ribosomal bases > 20%: 11 samples
# Filter 4 Outlier samples PCA: 10 samples
# Filter 5 Degraded RNA: 5 samples

```{r filters_5, echo=TRUE}
outlier_samples = c("MG-16-CER-RNA", "MG-08-CER-RNA", "13-095-GTS-unstim", "MG-15-SVZ-RNA", "MG-24-MGF-RNA", "14-046-GTS-unstim", "14-046-GFM-IL4", "16-117-THA-LPS", "16-074-GFM-unstim", "16-003-SVZ-TNFa")

Degraded_RNA = c("MG-24-HIP-RNA", "MG-12-CER-RNA", "15-089-GTS-unstim", "MG-15-SVZ-RNA", "MG-08-CER-RNA") 

list_filters = list(filter1 = star_alignment_1m, filter2 = mrna_mapping_5, filter3 = ribosomal_bases_20, filter4 = outlier_samples, filter5 = Degraded_RNA)
list_filters
samples2remove = unlist(list_filters)
unique(samples2remove)
```

### Upset plot

```{r filters_upset1, echo=TRUE, fig.width=6, fig.height=6, dpi=300 }
upset(fromList(list_filters), order.by = "freq")
```

# Filters
# Filter 1 < 5 M mapped reads: 21 samples
# Filter 2 mRNA mapping < 5%: 11 samples
# Filter 3 Ribosomal bases > 20%: 11 samples
# Filter 4 Degraded RNA: 5 samples

```{r filters_4, echo=TRUE}
Degraded_RNA = c("MG-24-HIP-RNA", "MG-12-CER-RNA", "15-089-GTS-unstim", "MG-15-SVZ-RNA", "MG-08-CER-RNA") 

list_filters = list(filter1 = star_alignment_5m, filter2 = mrna_mapping_5, filter3 = ribosomal_bases_20, filter4 = Degraded_RNA)
list_filters
samples2remove = unlist(list_filters)
unique(samples2remove)
```
 
### Upset plot
 
```{r filters_upset2, echo=TRUE, fig.width=6, fig.height=6, dpi=300 }
upset(fromList(list_filters), order.by = "freq")
```

### Filters
# filter 1: < 1 M reads
# filter 2: mRNA mapping < 5%
# filter 3: ribosomal bases > 20%
# filter 4: RNA degraded
# filter 5: duplicates
# filter 6: outlier (PCA)

```{r filters_6, echo=TRUE}
setwd("~/Documents/MiGASti/Databases")
Degraded_RNA = c("MG-24-HIP-RNA", "MG-12-CER-RNA", "15-089-GTS-unstim", "MG-15-SVZ-RNA", "MG-08-CER-RNA") 
duplicates =c("15-024-GFM_unstim", "16-074-GFM-unstim")
outliers =c("13-095-GTS-unstim")

list_filters = list(filter1 = star_alignment_1m, filter2 = mrna_mapping_5, filter3 = ribosomal_bases_20, filter4 = Degraded_RNA, filter5 = duplicates, filter6 = outliers)
list_filters
samples2remove = unlist(list_filters)
samples2remove
unique(samples2remove)
write.table(samples2remove, "samples2remove.txt")
```

### Upset plot
 
```{r filters_3, echo=TRUE, fig.width=6, fig.height=6, dpi=300 }
upset(fromList(list_filters), order.by = "freq")
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
